tests/integration_tests/functional/test_snapshot_basic.py

```diff
@@ -540,3 +544,41 @@ def test_vmgenid(guest_kernel_linux_6_1, rootfs, microvm_factory, snapshot_type)
 
         # Update the base for next iteration
         base_snapshot = snapshot
+
+
+# @pytest.mark.skipif(
+#    platform != "aarch64",
+#    reason="This is aarch64 specific test.",
+# )
+def test_physical_couter_reset_aarch64(uvm_nano, microvm_factory):
+    """
+    Test that the CNTPCT_EL0 register is reset on VM boot.
+    Because we cannot read the CNTVCT_EL0 (Virtual counter)
+    from python, we assume the smallesd VM will not consume more than
+    MAX_VALUE cycles to be created and snapshotted.
+    """
+    vm = uvm_nano
+    vm.add_net_iface()
+    vm.start()
+
+    snapshot = vm.snapshot_full()
+    vm.kill()
+    snap_editor = host.get_binary("snapshot-editor")
+
+    CNTPCT_EL0 = hex(0x603000000013DF01)
+    MAX_VALUE = 800_000_000
+
+    cmd = [
+        str(snap_editor),
+        "info-vmstate",
+        "vcpu-states",
+        "--vmstate-path",
+        str(snapshot.vmstate),
+    ]
+    _, stdout, _ = utils.check_output(cmd)
+    for line in stdout.splitlines():
+        s = line.split(" ")
+        if len(s) == 2:
+            reg_id, reg_value = line.split(" ")
```

Comment:
Why split again? this can be just `reg_id, reg_value = s` Can you put an example of the line format?

Reply:
nit: no need to split agin:

```
reg_id, reg_value = parts
```

Reply:
fixed